android-lib/src/main/java/com/kinvey/android/AsyncPullRequest.java

```diff
@@ -28,29 +29,55 @@
 /**
  * Class represents internal implementation of Async pull request that is used to create pull
  */
-public class AsyncPullRequest<T> extends AsyncClientRequest<KinveyPullResponse<T>> {
-    private final BaseDataStore store;
+public class AsyncPullRequest<T extends GenericJson> extends AsyncClientRequest<KinveyPullResponse<T>> {
+
+    private final BaseDataStore<T> store;
     private Query query;
+    private int pageSize;
+    private boolean isAutoPagination = false;
+
+    /**
+     * Async pull request constructor
+     * @param query Query that is used to fetch data from network
+     * @param store Kinvey data store instance to be used to execute network requests
+     * @param callback async callbacks to be invoked when job is done
+     */
+    public AsyncPullRequest(BaseDataStore<T>  store,
+                            Query query,
+                            KinveyPullCallback<T> callback) {
+        super(callback);
+        this.query = query;
+        this.store = store;
+    }
 
     /**
      * Async pull request constructor
      * @param query Query that is used to fetch data from network
+     * @param pageSize Page size for auto-pagination
      * @param store Kinvey data store instance to be used to execute network requests
      * @param callback async callbacks to be invoked when job is done
      */
-    public AsyncPullRequest(BaseDataStore store,
+    public AsyncPullRequest(BaseDataStore<T> store,
                             Query query,
-                            KinveyPullCallback<T> callback){
+                            int pageSize,
+                            KinveyPullCallback<T> callback) {
         super(callback);
         this.query = query;
         this.store = store;
+        this.isAutoPagination = true;
+        this.pageSize = pageSize;
```

Comment:
Do we need an `isAutoPagination` flag to indicate that auto-pagination is enabled if we have `pageSize` variable?

Reply:
I've removed `isAutoPagination` field. Currently we check that `pageSize > 0`.